custom_components/tahoma/sensor.py

```diff
@@ -104,35 +115,8 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
-        attribute = [
-            attr.value
-            for attr in self.device.attributes
-            if attr.name == CORE_MEASURED_VALUE_TYPE
-        ]
-        TEMPERATURE_UNIT = {
-            CORE_TEMPERATURE_IN_CELSIUS: TEMP_CELSIUS,
-            CORE_TEMPERATURE_IN_CELCIUS: TEMP_CELSIUS,
-            CORE_TEMPERATURE_IN_KELVIN: TEMP_KELVIN,
-            CORE_TEMPERATURE_IN_FAHRENHEIT: TEMP_FAHRENHEIT,
-        }.get(attribute[0], TEMP_CELSIUS,)
-        state = self.select_state(
-            CORE_CO2_CONCENTRATION_STATE,
-            CORE_CO_CONCENTRATION_STATE,
-            CORE_ELECTRIC_ENERGY_CONSUMPTION_STATE,
-            CORE_ELECTRIC_POWER_CONSUMPTION_STATE,
-            CORE_LUMINANCE_STATE,
-            CORE_RELATIVE_HUMIDITY_STATE,
-            CORE_TEMPERATURE_STATE,
-        )
-        return {
-            CORE_TEMPERATURE_STATE: TEMPERATURE_UNIT,
-            CORE_RELATIVE_HUMIDITY_STATE: UNIT_PERCENTAGE,
-            CORE_LUMINANCE_STATE: UNIT_LX,
-            CORE_ELECTRIC_POWER_CONSUMPTION_STATE: POWER_WATT,
-            CORE_ELECTRIC_ENERGY_CONSUMPTION_STATE: ENERGY_WATT_HOUR,
-            CORE_CO_CONCENTRATION_STATE: CONCENTRATION_PARTS_PER_MILLION,
```

Comment:
What about the units that can not be retrieved from the device attributes? It looks like this is removed in this PR?

Reply:
In such a case, we will return None. I don't introduce a regression. The previous code was not working. `select_state’ returns the value of a state. So in the case of my lux sensor, I got for instance 5356. So Celcius was always returned.
If you have other examples, I can easily add them.

Reply:
In that case it has been broken since the last update.. We should make sure we move back to the original behaviour where we fallback to the available state to decide the unit.

https://github.com/iMicknl/ha-tahoma/blame/c6efc4c82bb01651827329659413eac23ff5ea10/custom_components/tahoma/sensor.py#L119

Reply:
You are right, the previous refactor changed the behavior. I will fix it. I found another measured unit here: https://github.com/iMicknl/ha-tahoma/issues/80
I think the way I retrieved the units make more sense. Somfy seems consistent here.

Reply:
@tetienne I totally agree. Best would be to retrieve the value from the attributes. + a fallback to a hardcoded dict based on states that don't have those attributes.